test/functional/applications_controller_test.rb

```diff
@@ -95,6 +95,16 @@ class ApplicationsControllerTest < ActionController::TestCase
       assert_select ".release__badge", "AWS"
     end
 
+    should "show the deployment freeze badge" do
+      get :show, params: { id: @app.id }
+      assert_select ".release__badge", { text: "Do not deploy", count: 0 }
```

Comment:
```suggestion
      assert_select ".release__badge", false, text: "Do not deploy"
```

Minor: slightly more precise.

Reply:
So I tried this and it doesn't work because - it'll expects there to be no elements responding to the selector `.release_badge` (of which there is because of the AWS badge). The `text:` argument doesn't increase the specificity of the selector. To [perform more than one equality test](https://apidock.com/rails/ActionController/Assertions/SelectorAssertions/assert_select), you have to use the `:count` argument.

Reply:
Ah, OK. Good to know - thanks for trying :-)